### ISSUE TITLE
Support executing profiles from GitHub Enterprise

### ISSUE THREAD

**Issue by mcascone** [inspec/inspec#5284]
Please add support for running profiles stored in GitHub Enterprise.

## Detailed Description

Running `inspec exec https://github.mycompany.com/path/to/my/profile.tar.gz` results in the classic `Unrecognized content type: text/html; charset=utf-8. Assuming tar.gz
Error opening/processing profile.tar.gz: not in gzip format`

I can put it somewhere else that supports versioning - we are adopting ProGet - but would prefer to just use it at its source.

## Context

I need to be able to run version-controlled profiles from Jenkins, and using a git location would be easiest.

## Possible Implementation

This one's easy:

```powershell
inspec exec https://github.mycompany.com/path/to/my/profile.tar.gz --target winrm://my-target.mycompany.com --user <my user> --password <password> --controls=/my.*regex/
```

Aha! Link: https://chef.aha.io/features/SH-383

**Comment by james-stocks** [comment]
Yes at the moment InSpec needs the target URL to end with `.git`. 

Is `https://github.mycompany.com/path/to/my/profile.tar.gz` a tar file committed to git; or is it a release available for download from the Github UI?

**Comment by mcascone** [comment]
I’ve tried with both, and get the same text/html error. I tried the ‘raw’ link too; same result.

**Comment by aaronlippold** [comment]
we do this quite a bit 

You need to use the archieve tar.gz or .zip from the archieve/<branch>.targ.gz ...

**Comment by aaronlippold** [comment]
Also note, if your GH has auth setup and the repo is not public you need to cache our gh creds first and then do your inspec exec

**Comment by rlakey** [comment]
Same issue here but with Gitlab enterprise.

```
  - name: photon
    url: https://gitlaburl/photon-inspec-baseline/-/archive/development/photon-inspec-baseline-development.tar.gz

[2021-09-03T07:18:29-06:00] WARN: Unrecognized content type: text/html; charset=utf-8. Assuming tar.gz
Error opening/processing C:/Gitlab/wsoa-3.x-stig-inspec-baseline/vendor/cdaa91439a1288a354e268e80584d18715102e933a37cfb16d86e0377dd9b2c9.tar.gz: not in gzip format
```
